src/invidious/helpers/extractors.cr

```diff
@@ -49,7 +47,7 @@ private module Parsers
       # Instead, in its place is the amount of people currently watching. This behavior should be replicated
       # on Invidious once all features of livestreams are supported. On an unrelated note, defaulting to the current
       # time for publishing isn't a good idea.
```

Comment:
There is indeed a start timestamp for live videos:
![image](https://user-images.githubusercontent.com/52980486/129019650-d25c72dd-d85e-4bdf-af60-89cf10075cc0.png)

It's available as:
```
"microformat": {
  "playerMicroformatRenderer": {
    ...
    "liveBroadcastDetails": {
      "isLiveNow": true,
      "startTimestamp": "2020-02-22T20:14:45+00:00"
    },
    "uploadDate": "2020-02-22"
  }
},
...
```

Reply:
for a stream that ended:
```
"playerMicroformatRenderer": {
	...
  "publishDate": "2021-07-26",
	"liveBroadcastDetails": {
	"isLiveNow": false,
		"startTimestamp": "2021-07-26T07:44:26+00:00",
		"endTimestamp": "2021-07-27T00:58:16+00:00"
	},
	"uploadDate": "2021-07-26"
}
```

Reply:
Ah, nevermind. This parser never sees data from the `player` or `next` endpoints.

Reply:
> Ah, nevermind. This parser never sees data from the `player` or `next` endpoints.

Not yet at least! I'll be migrating all of the extraction logic we have in future PR(s) to the parsers. 

But even once that is implemented, I don't think we should be querying the `player` endpoint for every `videoRenderer` that represents a live-stream or premiere.